### ISSUE TITLE
Single line comments are merged on trailing trivia

### ISSUE THREAD

**Issue by OmarTawfik** [prettier/prettier#9959]
**Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEAHgBQCU+wAOlAL74D09+qOATgM4zUDcj+7cSFAAmIADQgIAB1zR2yUAENWrCAHcACsoTyUigDZrFAT3kSARq0VgA1nBgBlKdZxQA5shit0cCXAC25nDCwsEAMoru6IpucABiEKz+ijC47sggiugwEOIgABYw-voA6vk48OzOYHAOOhU4AG4Vxhlg7GYgrgKsMBpWbsnIqAYCEgBW7IQAQla29g6K-nBhrnDDo74gk4QOrm76cACK6BDwG-pjIM4ccKwZ5opB+nlSrK4wJTjCMPnIABwABgkbwgAhKVikGTecB6jXWEgAjqd4P1pLpMuwALRQODBYJ5VhwZFsOD9GJDJAjS5bAT+HCeby0-aHE5ndZUzYSGBPL4-P5IABM3KsOH0+wAwhB-JSQLCAKx5dACAAqT101KujR8AEkRAhHGB3jIAIIiBwwYyHC4CGg0IA)
```sh
--parser babel
```

**Input:**
```jsx
function x() {
} // first
; // second
```

**Output:**
```jsx
function x() {} // first // second

```

**Expected behavior:**
```jsx
function x() {} // first
// second
```

**Details:**

As you can see, the two comments (that should be on their own lines) are merged on a single line. I think this has to do with the `lineSuffix` command being used on two tokens on the same line. When that happens, both the suffixes are appended to the same line.

This is obviously very dangerous, and destroys the semantic meaning of the comment. Not only it makes it much harder to read and reason about (since comments cannot be broken), it also means that it will break any tools/parsers that read the comment contents and expect certain strings to be there/not there, like ignores and other directives.

| - | First | Second |
| --------------- | --------------- | --------------- |
| Before | ` first` | ` second` |
| After | ` first // second` | **removed** |

This does not just affect JavaScript, but it also affects all other languages, as they have no other means AFAICT of fixing this correctly using other available commands.

**Comment by OmarTawfik** [comment]
@sosukesuzuki I tracked the issue to this piece of code:

https://github.com/prettier/prettier/blob/74d144989659f0a03922f91c28d1f109cf12a14a/src/document/doc-printer.js#L468-L470

Happy to work on a fix and add tests, but since I'm a first time contributor, some pointers would be appreciated. Can you please let me know:

* Is there a case where `lineSuffix` can be used multiple times on the same line? I cannot find an existing example in a language.
* If not, should an unconditional break always exist after line suffixes?

TIA!

**Comment by fisker** [comment]
Actually, this is already fixed. Will post details.

**Comment by thorn0** [comment]
> Is there a case where lineSuffix can be used multiple times on the same line?

The Ruby plugin uses `lineSuffix` to support heredoc strings (https://github.com/prettier/plugin-ruby/pull/681):
```ruby
a(b(c(<<~HERE, <<~DOC)))
  this is the content of the first string
HERE
  this is the content of the second string
DOC
```

**Comment by OmarTawfik** [comment]
@fisker thanks for pointing out.
Do you plan on rebasing/merging #9772 since $9736 is now merged?
Going over your fix, it seems that this will also fix it for other languages as well.

**Comment by fisker** [comment]
I'm a little busy recently, and seems #9772 breaking some cases.

You can work on it if you like.

**Comment by OmarTawfik** [comment]
@fisker pulled the fix down and started going through the code. IIUC, I don't believe we are talking about the same issue. Your PR handles **multiple multi-line comments that should be on the same line**. This issue describes **single line comments that should NOT be on the same line**.

**Comment by fisker** [comment]
Just ignore my PR.

**Comment by thorn0** [comment]
fixed by #13438
